server/mscalendar/daily_summary.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package mscalendar
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/tz"
+	"github.com/pkg/errors"
+)
+
+const dailySummaryTimeWindow = time.Minute * 2
+
+var timeNowFunc = time.Now
+
+type DailySummary interface {
+	GetDailySummarySettingsForUser(user *User) (*store.DailySummarySettings, error)
+	SetDailySummaryPostTime(user *User, timeStr string) (*store.DailySummarySettings, error)
+	SetDailySummaryEnabled(user *User, enable bool) (*store.DailySummarySettings, error)
+	DailySummaryAll() error
+}
+
+func (m *mscalendar) GetDailySummarySettingsForUser(user *User) (*store.DailySummarySettings, error) {
+	dsumIndex, err := m.Store.LoadDailySummaryIndex()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dsum := range dsumIndex {
+		if dsum.MattermostUserID == user.MattermostUserID {
+			return dsum, nil
+		}
+	}
+
+	return nil, errors.New("No daily summary settings found")
+}
+
+func (m *mscalendar) SetDailySummaryPostTime(user *User, timeStr string) (*store.DailySummarySettings, error) {
+	t, err := time.Parse(time.Kitchen, timeStr)
+	if err != nil {
+		return nil, errors.New("Invalid time value")
+	}
+
+	if t.Minute() != 0 && t.Minute() != 30 {
+		return nil, errors.New("Time must be a multiple of 30 minutes.")
+	}
+
+	timezone, err := m.GetTimezone(user)
+	if err != nil {
+		return nil, err
+	}
+
+	dsumIndex, err := m.Store.LoadDailySummaryIndex()
```

Comment:
During these actions, shouldn't we use KVCompareAndSet to avoid problems with HA?

Reply:
Ticket here https://github.com/mattermost/mattermost-plugin-mscalendar/issues/47